frontend/src/metabase-lib/v1/expressions/index.ts

```diff
@@ -127,31 +127,60 @@ export function formatSegmentName(
  */
 export function parseDimension(
   name: string,
-  {
-    query,
-    stageIndex,
-    expressionIndex,
-  }: {
+  options: {
     query: Lib.Query;
     stageIndex: number;
-    source: string;
     expressionIndex: number | undefined;
+    startRule: string;
   },
 ) {
-  const columns = Lib.expressionableColumns(query, stageIndex, expressionIndex);
-
-  return columns.find(column => {
-    const displayInfo = Lib.displayInfo(query, stageIndex, column);
-
+  return getAvailableDimensions(options).find(({ info }) => {
     return EDITOR_FK_SYMBOLS.symbols.some(separator => {
       const displayName = getDisplayNameWithSeparator(
-        displayInfo.longDisplayName,
+        info.longDisplayName,
         separator,
       );
 
       return displayName === name;
     });
+  })?.dimension;
+}
+
+function getAvailableDimensions({
+  query,
+  stageIndex,
+  expressionIndex,
+  startRule,
+}: {
+  query: Lib.Query;
+  stageIndex: number;
+  expressionIndex: number | undefined;
+  startRule: string;
+}) {
+  const results = Lib.expressionableColumns(
+    query,
+    stageIndex,
+    expressionIndex,
+  ).map(dimension => {
+    return {
```

Comment:
I tried doing this:

```tsx
[
   ...Lib.expressionableColumns(/* ... */),
   ...Lib.availableMetric(/* ... */),
].map(dimenmsion => {
  const info = Lib.displayInfo(query, stageIndex, dimension);
  /* ... */
});
```

but TS complained about the `displayInfo` call.
That's why this function has 2 loops calling `Lib.displayInfo`